### ISSUE TITLE
Is there an equivalent of cmdlets *-AzureADServicePrincipalPolicy ?

### ISSUE THREAD

**Issue by Yvand** [microsoftgraph/msgraph-sdk-powershell#2343]
**Is your feature request related to a problem? Please describe.**
Module AzureADPreview has cmdlets [Get-AzureADServicePrincipalPolicy](https://learn.microsoft.com/en-us/powershell/module/azuread/get-azureadserviceprincipalpolicy?view=azureadps-2.0-preview) / [Add-AzureADServicePrincipalPolicy ](https://learn.microsoft.com/en-us/powershell/module/azuread/add-azureadserviceprincipalpolicy?view=azureadps-2.0-preview)/ [Remove-AzureADServicePrincipalPolicy](https://learn.microsoft.com/en-us/powershell/module/azuread/remove-azureadserviceprincipalpolicy?view=azureadps-2.0-preview).
I do not see their counterpart in Microsoft Graph PowerShell in the [cmdlet map page](https://learn.microsoft.com/en-us/powershell/microsoftgraph/azuread-msoline-cmdlet-map?view=graph-powershell-1.0).
Is it possible to achieve the same using Microsoft Graph PowerShell?
If not, do you have plans to implement them?

**Describe the solution you'd like**
My use case is to create a `TokenLifetimePolicy` as [documented here](https://learn.microsoft.com/en-us/azure/active-directory/saas-apps/sharepoint-on-premises-tutorial#configure-the-lifetime-of-the-security-token).

**Comment by SeniorConsulting** [comment]
Hey Yvant,

My first thoughts were "Weren't token lifetime policies deprecated?". I found my answer here:
https://learn.microsoft.com/en-us/azure/active-directory/develop/configurable-token-lifetimes#token-lifetime-policies-for-refresh-tokens-and-session-tokens

> As of January 30, 2021 you cannot configure refresh and session token lifetimes. Microsoft Entra no longer honors refresh and session token configuration in existing policies. New tokens issued after existing tokens have expired are now set to the [default configuration](https://learn.microsoft.com/en-us/azure/active-directory/develop/configurable-token-lifetimes#configurable-token-lifetime-properties). You can still configure access, SAML, and ID token lifetimes after the refresh and session token configuration retirement.

**Comment by Yvand** [comment]
I see, good catch, thank you for your insight @SeniorConsulting

**Comment by peombwa** [comment]
Thanks for following up on this.

The endpoints for managing a `tokenLifetimePolicy` for a service principal are indeed missing the service description (OpenAPI document) used to generate the SDKs. The missing paths are being tracked at https://github.com/microsoftgraph/msgraph-metadata/issues/322.

In the meantime, you can use `Invoke-MgGraphRequest` to assign a `tokenLifetimePolicy` for a service principal as shown below:
``` powershell
Invoke-MgGraphRequest -Method POST -Uri "v1.0/servicePrincipals/$SpId/tokenLifetimePolicies/`$ref" `
-Body @{"@odata.id" = "https://graph.microsoft.com/v1.0/policies/tokenLifetimePolicies/$TokenLifetimePolicyId" }
```

**Comment by Yvand** [comment]
@peombwa thank you, I tested your suggestion using Postman and it works like a charm!
Based on your example, I found [the API documentation](https://learn.microsoft.com/en-us/graph/api/application-post-tokenlifetimepolicies?view=graph-rest-1.0&tabs=http).

What is really not clear to me, is whether assigning a tokenLifetimePolicy to a service principal is still supported or not? Based on [this note](https://learn.microsoft.com/en-us/azure/active-directory/develop/configurable-token-lifetimes#token-lifetime-policies-for-refresh-tokens-and-session-tokens) it looks like it is not
